02_bubble/saurabhvyas9/main.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func bubble(s []int) []int {
+	r := make([]int, len(s))
+	copy(r, s)
+	countChecked := 1
+	for countChecked != 0 {
+		countChecked = 0
+		for i := 0; i < (len(r) - 1); i++ {
+			if r[i] > r[i+1] {
+				r[i], r[i+1] = r[i+1], r[i]
+				countChecked++
+			}
+		}
+	}
+	return r
+}
+
+func insertion(s []int) []int {
+	r := make([]int, len(s))
+	copy(r, s)
+	for i := 1; i < len(r); i++ {
+		temp := r[:i]
+		for j := 0; j <= len(temp)-1; j++ {
+			if temp[j] > r[i] {
+				temp[j], r[i] = r[i], temp[j]
+			}
+		}
+	}
+	return r
+}
+
+func main() {
+	fmt.Fprint(out, bubble([]int{3, 2, 1, 5}))
```

Comment:
Totally a matter of taste, but IMO this should go on it's own line with:
`fmt.Fprintln(out, bubble([]int{3, 2, 1, 5}))`

Reply:
Done.